### PR TITLE
Update reproducible comparing on linux

### LINKED ISSUE #3920: JDK23 Linux reproducible comparing tests failure
jdk23 linux reproducible comparing test failed with the diff https://ci.adoptium.net/job/Test_openjdk23_hs_special.system_x86-64_linux_testList_0/13/

```
        Total time: 28 seconds
        OpenJDK SBOM will be /home/jenkins/temurin-build/workspace/target/OpenJDK23U-sbom_x64_linux_hotspot_23_36-ea.json.
        Adding GLIBC version to SBOM: 2.17
        Adding GCC version to SBOM: 11.3.0
        BootJDK java : /usr/lib/jvm/jdk-22.0.2+9/bin/java
        Adding BOOTJDK to SBOM: 22.0.2+9
        Adding ALSA version to SBOM: 1.1.8
        FREETYPE_TO_USE=bundled
        Checking for FreeType include /home/jenkins/temurin-build/workspace/./build//src/src/java.desktop/share/native/libfreetype/include/freetype/freetype.h
        Found /home/jenkins/temurin-build/workspace/./build//src/src/java.desktop/share/native/libfreetype/include/freetype/freetype.h
        Checking and getting Tool Summary info:
        CycloneDX SBOM has been created in /home/jenkins/temurin-build/workspace/target/OpenJDK23U-sbom_x64_linux_hotspot_23_36-ea.json
        build.sh : 02:37:41 : All done!
        Comparing ...
        Only in /home/jenkins/jdkbinary/bin: jar.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jarsigner.debuginfo
        Only in /home/jenkins/jdkbinary/bin: java.debuginfo
        Only in /home/jenkins/jdkbinary/bin: javac.debuginfo
        Only in /home/jenkins/jdkbinary/bin: javadoc.debuginfo
        Only in /home/jenkins/jdkbinary/bin: javap.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jcmd.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jconsole.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jdb.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jdeprscan.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jdeps.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jfr.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jhsdb.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jimage.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jinfo.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jlink.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jmap.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jmod.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jpackage.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jps.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jrunscript.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jshell.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jstack.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jstat.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jstatd.debuginfo
        Only in /home/jenkins/jdkbinary/bin: jwebserver.debuginfo
        Only in /home/jenkins/jdkbinary/bin: keytool.debuginfo
        Only in /home/jenkins/jdkbinary/bin: rmiregistry.debuginfo
        Only in /home/jenkins/jdkbinary/bin: serialver.debuginfo
        Binary files /home/jenkins/jdkbinary/jmods/java.base.jmod and compare.1/jdk-23+36/jmods/java.base.jmod differ
        diff -r /home/jenkins/jdkbinary/lib/classlist compare.1/jdk-23+36/lib/classlist
        425d424
        < java/net/DefaultInterface
        428,430d426
        < java/net/Inet6Address
        < java/net/Inet6Address$Inet6AddressHolder
        < java/net/Inet6AddressImpl
        436,437d431
        < java/net/InterfaceAddress
        < java/net/NetworkInterface
        Only in /home/jenkins/jdkbinary/lib: jexec.debuginfo
        Only in /home/jenkins/jdkbinary/lib: jspawnhelper.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libattach.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libawt.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libawt_headless.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libawt_xawt.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libdt_socket.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libextnet.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libfontmanager.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libfreetype.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libinstrument.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libj2gss.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libj2pcsc.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libj2pkcs11.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libjaas.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libjava.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libjavajpeg.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libjawt.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libjdwp.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libjimage.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libjli.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libjsig.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libjsound.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libjsvml.debuginfo
        Only in /home/jenkins/jdkbinary/lib: liblcms.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libmanagement.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libmanagement_agent.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libmanagement_ext.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libmlib_image.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libnet.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libnio.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libprefs.debuginfo
        Only in /home/jenkins/jdkbinary/lib: librmi.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libsaproc.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libsctp.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libsimdsort.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libsplashscreen.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libsyslookup.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libverify.debuginfo
        Only in /home/jenkins/jdkbinary/lib: libzip.debuginfo
        Binary files /home/jenkins/jdkbinary/lib/modules and compare.1/jdk-23+36/lib/modules differ
        Binary files /home/jenkins/jdkbinary/lib/server/classes.jsa and compare.1/jdk-23+36/lib/server/classes.jsa differ
        Binary files /home/jenkins/jdkbinary/lib/server/classes_nocoops.jsa and compare.1/jdk-23+36/lib/server/classes_nocoops.jsa differ
        Only in /home/jenkins/jdkbinary/lib/server: libjvm.debuginfo
        diff -r /home/jenkins/jdkbinary/release compare.1/jdk-23+36/release
        12c12
        < BUILD_SOURCE_REPO="https://github.com/adoptium/temurin-build.git"
        ---
        > BUILD_SOURCE_REPO="https://github.com/adoptium/temurin-build"
        Differences found..., logged in: reprotest.diff
        -----------------------------------
        Rebuild_Same_JDK_Reproducibility_Test_0_FAILED
```

### PR DESCRIPTION
**Pull request by sophia-guo**
Fix #3920 

Using common compare script
Remove non-revolant files
Remove debuginfo if exists from aqa-tests level

### CODE REVIEW

**Comment by sophia-guo** [comment]
https://ci.adoptium.net/job/Test_openjdk23_hs_special.system_x86-64_linux_testList_0/16/console

File: tooling/reproducible/repro_common.sh [hunk 1]
```diff
@@ -327,14 +327,18 @@ function processModuleInfo() {
...
-  if [[ "$OS" =~ CYGWIN* ]] || [[ "$OS" =~ Darwin* ]]; then
-    rm -rf "$JDK_DIR/bin/server/classes.jsa"
-    rm -rf "$JDK_DIR/bin/server/classes_nocoops.jsa"
+  rm -rf "$JDK_DIR/bin/server/classes.jsa"
+  rm -rf "$JDK_DIR/bin/server/classes_nocoops.jsa"
+  rm -rf "$JDK_DIR/release"
+  rm -rf "$JDK_DIR/NOTICE"  
+
+  if [[ "$OS" =~ Linux ]]; then
+    find "$JDK_DIR/" -type f -name "*.debuginfo" -delete
```
Comment (andrew-m-leonard) [inline tooling/reproducible/repro_common.sh:341]: I don't think removing the debuginfo here is the right thing to do, we should ensure only the JDK tarball is unpacked

Also, the "release" file should not be totally ignored, only the specific Temurin lines should be removed, as in cleanTemurinBuildInfo()

**Review by andrew-m-leonard — CHANGES_REQUESTED** [review]

**Comment by sophia-guo** [comment]
https://ci.adoptium.net/job/Test_openjdk23_hs_special.system_x86-64_linux_testList_0/27/

File: tooling/reproducible/linux_repro_build_compare.sh [hunk 2]
```diff
@@ -185,41 +180,36 @@ if [ -z "$JDK_PARAM" ] && [ ! -d "jdk-${TEMURIN_VERSION}" ] ; then
...
   tar xpfz "$JDK_PARAM" --strip-components=1 -C "$PWD/jdk-${TEMURIN_VERSION}"
 else
-  echo "Local jdk dir"
-  isJdkDir=true
-fi
-
-comparedDir="jdk-${TEMURIN_VERSION}"
-if [ "${isJdkDir}" = true ]; then
-  comparedDir=$JDK_PARAM
+  #Local jdk dir
```
Comment (karianna) [inline tooling/reproducible/linux_repro_build_compare.sh:190]:
```suggestion
  # Local JDK dir
```

**Review by karianna — APPROVED** [review]

File: tooling/reproducible/repro_common.sh [hunk 3]
```diff
@@ -327,15 +327,24 @@ function processModuleInfo() {
...
+#  classes.jsa, classes_nocoops.jsa - CDS archive caches will differ due to Vendor string differences
+function removeExcludedFiles() {
   local JDK_DIR="$1"
-  local OS="$2"
-
-  if [[ "$OS" =~ CYGWIN* ]] || [[ "$OS" =~ Darwin* ]]; then
-    rm -rf "$JDK_DIR/bin/server/classes.jsa"
-    rm -rf "$JDK_DIR/bin/server/classes_nocoops.jsa"
-  fi
+  excluded="NOTICE cacerts classlist classes.jsa classes_nocoops.jsa"
```
Comment (andrew-m-leonard) [inline tooling/reproducible/repro_common.sh:337]: This is not quite right, the CDS archives will only differ on Windows and Mac, they "should be" identical on Linux.

Also, when comparing Temurin builds, the NOTICE and cacerts will be identical and should not be excluded.

I suspect you've added the CDS files for Linux due to the classlist ipv6 difference, that is a known issue, and we need to ensure we have the same ipv6 setting on the reproduce VM.

File: tooling/reproducible/repro_common.sh [hunk 4]
```diff
@@ -421,9 +430,6 @@ function cleanTemurinFiles() {
 
   echo "Cleaning Temurin build-scripts specific files and metadata from ${DIR}"
 
-  echo "Removing Temurin NOTICE file from $DIR"
-  rm "${DIR}"/NOTICE
```
Comment (andrew-m-leonard) [inline tooling/reproducible/repro_common.sh:425]: This should remain, cleanTemurinFiles() is called when comparing a Temurin build against a plain OpenJDK build, in which case Temurin specific metadata needs cleaning.

In this testcase scenario we are comparing Temurin vs Temurin

File: tooling/reproducible/repro_common.sh [hunk 5]
```diff
@@ -454,9 +460,6 @@ function cleanTemurinFiles() {
     sed -i '/^SOURCE=.*$/d' "${DIR}/release"
   fi
 
-  echo "Removing cacerts file, as Temurin builds with different Mozilla cacerts"
-  find "${DIR}" -type f -name "cacerts" -delete
```
Comment (andrew-m-leonard) [inline tooling/reproducible/repro_common.sh:458]: Ditto, should remain here

File: tooling/reproducible/repro_process.sh [hunk 6]
```diff
@@ -23,7 +23,7 @@ OS="$2"
 
 expandJDK "$JDK_DIR" "$OS"
 
-removeGeneratedClasses "$JDK_DIR" "$OS"
+removeExcludedFiles "$JDK_DIR"
```
Comment (andrew-m-leonard) [inline tooling/reproducible/repro_process.sh:26]: this was called GeneratedClasses, as it related to just cleaning the generated CDS artifacts

**Review by andrew-m-leonard — CHANGES_REQUESTED** [review]

File: tooling/reproducible/repro_common.sh (hunk 3 above)
Comment (sophia-guo) [reply tooling/reproducible/repro_common.sh]: Updated @andrew-m-leonard

**Comment by sophia-guo** [comment]
https://ci.adoptium.net/job/Test_openjdk23_hs_special.system_x86-64_linux_testList_0/29/console

**Review by andrew-m-leonard — APPROVED** [review]
looks good thanks Sophia

**Pull request merged**